backend/src/reservations/reservations.controller.ts

```diff
@@ -65,6 +65,27 @@ export class ReservationsController {
     }
   }
 
+  @Post('/mark-complete')
+  @UsePipes(new ValidationPipe())
+  async mark_complete(@Res() response, @Body() id: number) {
+    try {
+      const reservation = await this.reservationsService.updatePaymentStatus(
+        id,
+        true,
+      );
+      if (!reservation.payment_completed) {
+        return response
+          .status(422)
+          .json(`Failed to update reservation as completed`);
+      }
+      return response.status(201).json(reservation);
```

Comment:
I think the response status should be 200 here, since the status 201 means that a new record was created, which doesn't seem to be the case here?